AudioSinkExample/AudioSinks/ExampleAudioRecorder.m

```diff
@@ -0,0 +1,143 @@
+@property (nonatomic, strong) AVAssetWriterInput *audioRecorderInput;
+@property (nonatomic, assign) CMTime recorderTimestamp;
+@property (nonatomic, assign) int numberOfChannels;
+
+@property (nonatomic, weak) TVIAudioTrack *audioTrack;
+
+@end
+
+@implementation ExampleAudioRecorder
+
+- (instancetype)initWithAudioTrack:(TVIAudioTrack *)audioTrack identifier:(NSString *)identifier {
+    self = [super init];
+    if (self) {
+        _recorderTimestamp = kCMTimeInvalid;
+
+        [self startRecordingAudioTrack:audioTrack withIdentifier:identifier];
+    }
+    return self;
+}
+
+- (void)startRecordingAudioTrack:(TVIAudioTrack *)audioTrack withIdentifier:(NSString *)identifier {
+    NSParameterAssert(audioTrack);
+    NSParameterAssert(identifier);
+
+    // Setup Recorder
+    NSError *error = nil;
+    _audioRecorder = [[AVAssetWriter alloc] initWithURL:[[self class] recordingURLWithIdentifier:identifier]
+                                               fileType:AVFileTypeWAVE
+                                                  error:&error];
+
+    if (error) {
+        NSLog(@"Error setting up audio recorder: %@", error);
+        return;
+    }
+
+    // The iOS audio device captures in mono.
+    // The mixer produces stereo audio for each remote Participant, even if they send mono audio.
+    _numberOfChannels = [audioTrack isKindOfClass:[TVILocalAudioTrack class]] ? 1 : 2;
+
+    // Assume that TVIAudioTrack will produce interleaved stereo LPCM @ 16-bit / 48khz
+    NSDictionary<NSString *, id> *outputSettings = @{AVFormatIDKey : @(kAudioFormatLinearPCM),
+                                                     AVSampleRateKey : @(48000),
+                                                     AVNumberOfChannelsKey : @(self.numberOfChannels),
+                                                     AVLinearPCMBitDepthKey : @(16),
+                                                     AVLinearPCMIsFloatKey : @(NO),
+                                                     AVLinearPCMIsBigEndianKey : @(NO),
+                                                     AVLinearPCMIsNonInterleaved : @(NO)};
+
+    _audioRecorderInput = [AVAssetWriterInput assetWriterInputWithMediaType:AVMediaTypeAudio outputSettings:outputSettings];
+    _audioRecorderInput.expectsMediaDataInRealTime = YES;
+
+    if ([_audioRecorder canAddInput:_audioRecorderInput]) {
+        [_audioRecorder addInput:_audioRecorderInput];
+        BOOL success = [_audioRecorder startWriting];
+
+        if (success) {
+            NSLog(@"Started recording audio track to: %@", _audioRecorder.outputURL);
+            [audioTrack addSink:self];
+            _audioTrack = audioTrack;
+            _identifier = identifier;
+        } else {
+            NSLog(@"Couldn't start the AVAssetWriter: %@ error: %@", _audioRecorder, _audioRecorder.error);
+        }
+    }
+
+    // TODO: CE - Backgrounding support. We could start a task here, and handle cleanup if it expires.
```

Comment:
Didn't add backgrounding support. I'm not sure if we want to take the effort to do this or not. If not, we could acknowledge that this would be a good place to start a background task with a comment.

Reply:
Great! Will update the comment.